cuda_core/tests/test_program.py

```diff
@@ -150,3 +248,55 @@ def test_program_close():
     program = Program(code, "c++")
     program.close()
     assert program.handle is None
+
+
+@nvvm_available
+def test_nvvm_deferred_import():
+    """Test that our deferred NVVM import works correctly"""
+    from cuda.core.experimental._program import _get_nvvm_module
+
+    nvvm = _get_nvvm_module()
+    assert nvvm is not None
+
+
+@nvvm_available
+def test_nvvm_program_creation(nvvm_ir):
+    """Test basic NVVM program creation"""
+    program = Program(nvvm_ir, "nvvm")
+    assert program.backend == "NVVM"
+    assert program.handle is not None
+    program.close()
+
+
+@nvvm_available
+def test_nvvm_compile_invalid_target(nvvm_ir):
+    """Test that NVVM programs reject invalid compilation targets"""
+    program = Program(nvvm_ir, "nvvm")
+    with pytest.raises(ValueError, match='NVVM backend only supports target_type="ptx"'):
+        program.compile("cubin")
+    program.close()
+
+
+@nvvm_available
+@pytest.mark.parametrize(
+    "options",
+    [
+        ProgramOptions(name="test1", arch="sm_90", device_code_optimize=False),
+        ProgramOptions(name="test2", arch="sm_100", device_code_optimize=False),
+        ProgramOptions(name="test3", arch="sm_110", device_code_optimize=True),
```

Comment:
I think sm_110 is not supported by 12.x. Let's make it skipped based on libnvvm version. A similar approach can be found here:
https://github.com/NVIDIA/cuda-python/blob/d64120bb6733391989174456daaf18643ea29711/cuda_core/tests/test_launcher.py#L279-L285